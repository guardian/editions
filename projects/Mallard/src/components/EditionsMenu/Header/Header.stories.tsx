import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { withKnobs, color, text } from '@storybook/addon-knobs'
import { EditionsMenuHeader } from './Header'

storiesOf('EditionsMenuHeader', module)
    .addDecorator(withKnobs)
    .add('EditionsMenuHeader - default', () => (
        <EditionsMenuHeader>
            {text('Header Text', 'Regions')}
        </EditionsMenuHeader>
    ))
