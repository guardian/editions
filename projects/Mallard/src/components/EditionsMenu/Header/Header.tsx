import React from 'react';
import { StyleSheet, View } from 'react-native';
import { TitlepieceText } from 'src/components/styled-text';
import { color } from 'src/theme/color';

const styles = StyleSheet.create({
	container: {
		flex: 1,
		marginBottom: 4,
	},
	shadowBox: {
		backgroundColor: 'white',
		paddingBottom: 38,
		paddingLeft: 96,
		paddingTop: 4,
		shadowColor: color.palette.neutral[7],
		shadowOffset: { width: 0, height: 4 },
		shadowOpacity: 0.1,
		shadowRadius: 4,
	},
	text: { fontSize: 24, lineHeight: 30 },
});

const EditionsMenuHeader = ({ children }: { children: string }) => (
	<View style={styles.container}>
		<View style={styles.shadowBox}>
			<TitlepieceText style={styles.text}>{children}</TitlepieceText>
		</View>
	</View>
);

export { EditionsMenuHeader };
