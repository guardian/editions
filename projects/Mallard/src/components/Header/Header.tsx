import { useNavigation } from '@react-navigation/native';
import React from 'react';
import { Dimensions, StyleSheet, View } from 'react-native';
import { isTablet } from 'react-native-device-info';
import { Button } from '../Button/Button';
import { CloseButton } from '../Button/CloseButton';
import { IssueTitle } from '../issue/issue-title';
import { Header } from '../layout/header/header';

const width = Dimensions.get('window').width; //full width
const height = Dimensions.get('window').height; //full height

const ModalStyles = StyleSheet.create({
	wrapper: {
		alignItems: 'center',
		flexDirection: 'column',
		justifyContent: 'center',
		flex: 1,
	},
	container: {
		minHeight: isTablet() ? 600 : height,
		width: isTablet() ? 400 : width,
	},
});

const HeaderScreenContainer = ({
	actionLeft,
	actionRight = false,
	children,
	title,
}: {
	actionLeft: boolean;
	actionRight?: boolean;
	children: React.ReactNode;
	title: string;
}) => {
	const navigation = useNavigation();
	return (
		<View style={ModalStyles.wrapper}>
			<View style={ModalStyles.container}>
				<Header
					leftAction={
						actionLeft ? (
							<Button
								icon={'\uE00A'}
								alt="Back"
								onPress={() => navigation.goBack()}
							></Button>
						) : null
					}
					action={
						actionRight ? (
							<CloseButton
								accessibilityLabel={`Close the ${title} screen`}
								accessibilityHint="Closes the current screen"
								onPress={() => navigation.goBack()}
							/>
						) : null
					}
					layout={'center'}
				>
					<IssueTitle title={title} />
				</Header>
				{children}
			</View>
		</View>
	);
};

export { HeaderScreenContainer };
