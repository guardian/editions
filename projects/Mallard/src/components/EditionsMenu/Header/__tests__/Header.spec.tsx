import React from 'react';
import type { ReactTestRendererJSON } from 'react-test-renderer';
import TestRenderer from 'react-test-renderer';
import { EditionsMenuHeader } from '../Header';

describe('EditionsMenuHeader', () => {
	it('should display a default EditionsMenuHeader with correct styling', () => {
		const component: ReactTestRendererJSON | null = TestRenderer.create(
			<EditionsMenuHeader>Special Editions</EditionsMenuHeader>,
		).toJSON();
		expect(component).toMatchSnapshot();
	});
});
