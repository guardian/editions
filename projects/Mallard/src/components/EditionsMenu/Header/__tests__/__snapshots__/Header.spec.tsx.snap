// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditionsMenuHeader should display a default EditionsMenuHeader with correct styling 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingBottom": 38,
      "paddingLeft": 96,
      "paddingTop": 4,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "flexShrink": 0,
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 30,
          "lineHeight": 33,
        },
        Object {
          "fontSize": 24,
          "lineHeight": 30,
        },
      ]
    }
  >
    Special Editions
  </Text>
</View>
`;
